pallets/funding/src/lib.rs

```diff
@@ -1280,54 +1325,115 @@ pub mod pallet {
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(now: BlockNumberFor<T>) -> Weight {
 			// Get the projects that need to be updated on this block and update them
+			let mut used_weight = Weight::from_parts(0, 0);
 			for (project_id, update_type) in ProjectsToUpdate::<T>::take(now) {
 				match update_type {
 					// EvaluationRound -> AuctionInitializePeriod | EvaluationFailed
 					UpdateType::EvaluationEnd => {
-						unwrap_result_or_skip!(Self::do_evaluation_end(project_id), project_id, |e| e);
+						used_weight = used_weight.saturating_add(
+							unwrap_result_or_skip!(
+								Self::do_evaluation_end(project_id),
+								project_id,
+								|e: DispatchErrorWithPostInfo<PostDispatchInfo>| { e.error }
+							)
+							.actual_weight
+							.unwrap_or_default(),
```

Comment:
`.unwrap_or_default()` caused few problems before. Since this "default" value (what is the default in this case?) is in a very sensible part of the logic, I would handle the case of failure better (even though it should never actually happen)

Reply:
I get your point. We know from looking at the code that `actual_weight` will never be None, but still we can set the fallback case to be the worst case benchmark

Reply:
check af9de40e3bb50ba5f01c786bf577ed900b17806d